infra-cdk/lib/stacks/ReportsTableStack.ts

```diff
@@ -0,0 +1,54 @@
+import * as cdk from 'aws-cdk-lib';
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+import { Construct } from 'constructs';
+import { Stage, Tenant } from '../core/enums';
+import { BaseStackProps } from '../core/types';
+
+const BASE_TABLE_NAME = 'ReportsTable';
+
+/** DynamoDB table storing reports */
+export class ReportsTableStack extends cdk.Stack {
+    private stage: Stage;
+    private tenant: Tenant;
+
+    constructor(scope: Construct, id: string, props: BaseStackProps) {
+        super(scope, id, props);
+
+        this.stage = props.stage;
+        this.tenant = props.tenant;
+
+        const reportsTable = new dynamodb.Table(
+            this,
+            `ReportsTable-${props.stage}-${props.tenant}`,
+            {
+                tableName: this.getTableName(),
+                partitionKey: { name: 'ReportID', type: dynamodb.AttributeType.STRING },
+                removalPolicy: cdk.RemovalPolicy.RETAIN,
+            }
+        );
+
+        // // Add additional attributes to the table
```

Comment:
I created new "IReport.ts" file to add attributes for the ReportsTable. I decided to add these attributes in this file instead of directly adding them to ReportsTableStack.ts file since this way is safer and no bugs.

Reply:
> We decided we can uncomment these, especially if it's useful :) Better not to leave commented code

I created new "IReport.ts" file to add attributes for the ReportsTable. I decided to add these attributes in this file instead of directly adding them to ReportsTableStack.ts file since this way is safer and no bugs.